src/otx/core/model/entity/base.py

```diff
@@ -170,41 +173,67 @@ def map_class_names(src_classes: list[str], dst_classes: list[str]) -> list[int]
                 src2dst.append(-1)
         return src2dst
 
-    def export(self, output_dir: Path, export_format: OTXExportFormat) -> None:
+    def export(
+        self,
+        output_dir: Path,
+        export_format: OTXExportFormatType,
+        precision: OTXExportPrecisionType = OTXExportPrecisionType.FP32,
+    ) -> Path:
         """Export this model to the specified output directory.
 
         Args:
-            output_dir: Directory path to save exported binary files.
-            export_format: Format in which this `OTXModel` is exported.
+            output_dir (Path): directory for saving the exported model
+            export_format (OTXExportFormatType): format of the output model
+            precision (OTXExportPrecisionType): precision of the output model
+        Returns:
+            Path: path to the exported model.
         """
-        if export_format == OTXExportFormat.OPENVINO:
-            self._export_to_openvino(output_dir)
-        if export_format == OTXExportFormat.ONNX:
-            self._export_to_onnx()
-        if export_format == OTXExportFormat.EXPORTABLE_CODE:
-            self._export_to_exportable_code()
+        exporter = self._create_exporter()
+        metadata = self._generate_model_metadata()
 
-    def _export_to_openvino(self, output_dir: Path) -> None:
-        """Export to OpenVINO Intermediate Representation format.
+        if export_format == OTXExportFormatType.OPENVINO:
+            return exporter.to_openvino(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.ONNX:
+            return exporter.to_onnx(self.model, output_dir, self._EXPORTED_MODEL_BASE_NAME, precision, metadata)
+        if export_format == OTXExportFormatType.EXPORTABLE_CODE:
+            return self._export_to_exportable_code()
```

Comment:
Why don't we hide this branching into `exporter` as well?
```suggestion
        return exporter.export(exgport_format, ...)
```

Reply:
Same answer

Reply:
Yes, Kirill has already done that in his PR which would be merged after this one